pkg/imgpkg/bundle/fetch_images.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package bundle
+
+import (
+	"fmt"
+
+	regname "github.com/google/go-containerregistry/pkg/name"
+	"github.com/vmware-tanzu/carvel-imgpkg/pkg/imgpkg/internal/util"
+	"github.com/vmware-tanzu/carvel-imgpkg/pkg/imgpkg/lockconfig"
+)
+
+// FetchAllImagesRefs returns a flat list of nested bundles and every image reference for a specific bundle
+func (o *Bundle) FetchAllImagesRefs(concurrency int, ui util.UIWithLevels, sigFetcher SignatureFetcher) ([]*Bundle, error) {
```

Comment:
util.* isnt accessible to external users (can we have a separate package under pkg/imgpkg/ so that we arent doing this accidently)